In [1]:
import pandas as pd



Q1: What is the percentage of people at each age?

In [12]:
rawDF = pd.read_csv("Exercise Survey.csv")

rawDF.head()

Timestamp Username First and Last Name:  How old are you?  \
0  2024/01/24 2:11:49 PM EST      NaN     cydney kapperman                21   
1  2024/01/24 2:30:05 PM EST      NaN        Allee Mainord                22   
2  2024/01/24 2:47:44 PM EST      NaN         jess herring                21   
3  2024/01/24 2:50:10 PM EST      NaN           Paige Culp                22   
4  2024/01/24 3:24:49 PM EST      NaN        Maggie Eddins                21   

              What type of exercises do you do?   \
0                                Lifting;Running   
1                                Lifting;Walking   
2                                Running;Walking   
3                         Lifting;Playing Sports   
4  Lifting;Running;Biking;Playing Sports;Walking   

   How many days out of the week do you exercise?  \
0                                               3   
1                                               4   
2                                               4   
3                                               2   
4                                               5   

  What time of day do you normally exercise?  \
0                         12:00 pm - 6:00 pm   
1                         6:00 am - 12:00 pm   
2                         12:00 pm - 6:00 pm   
3                         6:00 am - 12:00 pm   
4                         6:00 pm - 12:00 am   

                       Where do you exercise?  \
0                    Cage Center;Viking Trail   
1                    Cage Center;Richards Gym   
2                                Viking Trail   
3  Cage Center;Richards Gym;Intramural Fields   
4  Cage Center;Richards Gym;Intramural Fields   

  What feelings are you having that makes you want to exercise?  \
0                                      Anxious;Angry              
1                                               Calm              
2                       Happy;Anxious;Sad;Angry;Calm              
3                                              Happy              
4                                   Happy;Angry;Calm              

  Do you listen to music when you exercise?  \
0                                       Yes   
1                                       Yes   
2                                       Yes   
3                                 Sometimes   
4                                       Yes   

         Why do you exercise at the chosen location?  \
0  Easily Accessible;Has equipment;Friends exerci...   
1                    Easily Accessible;Has equipment   
2  Easily Accessible;Great Scenery;Friends exerci...   
3                    Easily Accessible;Has equipment   
4  Easily Accessible;Has equipment;Friends exerci...   

  Do you think listening to music while exercising helps relieve the negative feelings?  
0                                                Yes                                     
1                                                Yes                                     
2                                                Yes                                     
3                                                Yes                                     
4                                                Yes

In [24]:
set( rawDF["How old are you?"].tolist() )

{19, 20, 21, 22, 29}

In [7]:
rawDFNum = len(rawDF)

rawDFNum

32

In [8]:
age19DF = len(rawDF[rawDF['How old are you?'] == 19])

In [11]:
(age19DF / rawDFNum) * 100

6.25

6.25% of people are 19.

In [14]:
age20DF = len(rawDF[rawDF['How old are you?'] == 20])

age20DF

8

In [15]:
(age20DF / rawDFNum) * 100

25.0

25% of people are 20 years old.

In [16]:
age21DF = len(rawDF[rawDF['How old are you?'] == 21])

age21DF

17

In [17]:
(age21DF / rawDFNum) * 100

53.125

53.1% of people are 21 years old

In [18]:
age22DF = len(rawDF[rawDF['How old are you?'] == 22])

age22DF

4

In [19]:
(age22DF / rawDFNum) * 100

12.5

12.5% of people are 22 years old.

In [26]:
age29DF = len(rawDF[rawDF['How old are you?'] == 29])

age29DF

1

In [27]:
(age29DF / rawDFNum) * 100

3.125

3.1% of people are 29 years old.

Q2: What are the percentages of each exercise done?

In [31]:
set( rawDF["What type of exercises do you do? "].tolist() )

{'Biking',
 'Lifting;Biking;Walking',
 'Lifting;Inclined Treadmill',
 'Lifting;Playing Sports',
 'Lifting;Playing Sports;Walking',
 'Lifting;Playing Sports;Walking;Pilates, zumba',
 'Lifting;Playing Sports;Yoga',
 'Lifting;Running',
 'Lifting;Running;Biking;Playing Sports;Walking',
 'Lifting;Running;Biking;Playing Sports;Walking;Swimming',
 'Lifting;Running;Calisthenics',
 'Lifting;Running;Playing Sports',
 'Lifting;Running;Playing Sports;Walking',
 'Lifting;Running;Playing Sports;Yoga',
 'Lifting;Running;Walking',
 'Lifting;Running;Walking;High Intensity Interval Training',
 'Lifting;Running;Yoga',
 'Lifting;Walking',
 'Playing Sports;Walking',
 'Running;Playing Sports',
 'Running;Walking'}

In [72]:
liftingDF = len(rawDF[rawDF['What type of exercises do you do? '].str.contains('Lifting')])

liftingDF

27

In [73]:
(liftingDF / rawDFNum) * 100

84.375

84.375% of people lift for exercise.

In [74]:
runningDF = len(rawDF[rawDF['What type of exercises do you do? '].str.contains('Running')])

runningDF

18

In [75]:
(runningDF / rawDFNum) * 100

56.25

56.25% of people run for exercise.

In [76]:
biking = len(rawDF[rawDF['What type of exercises do you do? '].str.contains('Biking')])

biking

5

In [77]:
(biking / rawDFNum) * 100

15.625

15.625% of people bike for exercise.

In [78]:
sports = len(rawDF[rawDF['What type of exercises do you do? '].str.contains('Playing Sports')])

sports

20

In [79]:
(sports / rawDFNum) * 100

62.5

62.5% of people play sports for exercise.

In [80]:
walking = len(rawDF[rawDF['What type of exercises do you do? '].str.contains('Walking')])

walking

17

In [81]:
(walking / rawDFNum) * 100

53.125

53.125% of people walk for exercise.

In [82]:
swimming = len(rawDF[rawDF['What type of exercises do you do? '].str.contains('Swimming')])

swimming

2

In [83]:
(swimming / rawDFNum) * 100

6.25

6.25% of people swim for exercise.

Q3: What percentage of people exercise for how many days?

In [36]:
set( rawDF["How many days out of the week do you exercise?"].tolist() )

{2, 3, 4, 5, 6, 7}

In [38]:
exc2Days = len(rawDF[rawDF["How many days out of the week do you exercise?"] == 2])

exc2Days

1

In [39]:
(exc2Days / rawDFNum) * 100

3.125

3.1% of people exercise 2 days out of the week.

In [40]:
exc3Days = len(rawDF[rawDF["How many days out of the week do you exercise?"] == 3])

exc3Days

4

In [41]:
(exc3Days / rawDFNum) * 100

12.5

12.5% of people exercise 3 days out of the week.

In [42]:
exc4Days = len(rawDF[rawDF["How many days out of the week do you exercise?"] == 4])

exc4Days

6

In [43]:
(exc4Days / rawDFNum) * 100

18.75

18.75% of people exercise 4 days out of the week.

In [44]:
exc5Days = len(rawDF[rawDF["How many days out of the week do you exercise?"] == 5])

exc5Days

10

In [46]:
(exc5Days / rawDFNum) * 100

31.25

31.25% of people exercise 5 days out of the week.

In [47]:
exc6Days = len(rawDF[rawDF["How many days out of the week do you exercise?"] == 6])

exc6Days

8

In [48]:
(exc6Days / rawDFNum) * 100

25.0

25% of people exercise 6 days out of the week.

In [49]:
exc7Days = len(rawDF[rawDF["How many days out of the week do you exercise?"] == 7])

exc7Days

3

In [50]:
(exc7Days / rawDFNum) * 100

9.375

9.38% of people exercise 7 days out of the week.

Q4: What percent of people exercise at the selected time?

In [51]:
set( rawDF["What time of day do you normally exercise?"].tolist() )

{'12:00 pm - 6:00 pm', '6:00 am - 12:00 pm', '6:00 pm - 12:00 am'}

In [54]:
midTime = len(rawDF[rawDF["What time of day do you normally exercise?"] == "12:00 pm - 6:00 pm"])

midTime

17

In [56]:
(midTime / rawDFNum) * 100

53.125

53.125% of people workout in the middle of the day.

In [57]:
earlyTime = len(rawDF[rawDF["What time of day do you normally exercise?"] == "6:00 am - 12:00 pm"])

earlyTime

8

In [58]:
(earlyTime / rawDFNum) * 100

25.0

25% of people workout early in the morning.

In [59]:
nightTime = len(rawDF[rawDF["What time of day do you normally exercise?"] == "6:00 pm - 12:00 am"])

nightTime

7

In [60]:
(nightTime / rawDFNum) * 100

21.875

21.875% of people workout later in the night.

Q5: What percentage of people exercises at a specific location?

In [84]:
cage = len(rawDF[rawDF['Where do you exercise?'].str.contains('Cage Center')])

cage

24

In [85]:
(cage / rawDFNum) * 100

75.0

75% of people exercises at the Cage Center.

In [86]:
vtrail = len(rawDF[rawDF['Where do you exercise?'].str.contains('Viking Trail')])

vtrail

13

In [87]:
(vtrail / rawDFNum) * 100

40.625

40.625% of people exercises on Viking Trail.

In [88]:
richards = len(rawDF[rawDF['Where do you exercise?'].str.contains('Richards Gym')])

richards

12

In [89]:
(richards / rawDFNum) * 100

37.5

37.5% of people exercises at Richards Gym.

In [90]:
intra = len(rawDF[rawDF['Where do you exercise?'].str.contains('Intramural Fields')])

intra

7

In [91]:
(intra / rawDFNum) * 100

21.875

21.875% of people exercises at Intramural Fields.

In [92]:
offCampus = len(rawDF[rawDF['Where do you exercise?'].str.contains('Off Campus')])

offCampus

6

In [93]:
(offCampus / rawDFNum) * 100

18.75

18.75% of people exercises off campus somewhere.

In [97]:
mountain = len(rawDF[rawDF['Where do you exercise?'].str.contains('Mountain Campus')])

mountain

2

In [95]:
(mountain / rawDFNum) * 100

6.25

6.25% of people exercises on Mountain campus.

Q6: Percentage of people who have each feeling out of the choices when exercising.

In [98]:
happy = len(rawDF[rawDF['What feelings are you having that makes you want to exercise?'].str.contains('Happy')])

happy

27

In [99]:
(happy / rawDFNum) * 100

84.375

84.375% of people feel like being happy makes them want to exercise.

In [100]:
anxious = len(rawDF[rawDF['What feelings are you having that makes you want to exercise?'].str.contains('Anxious')])

anxious

15

In [101]:
(anxious / rawDFNum) * 100

46.875

46.875% of people feel like being anxious makes them want to exercise.

In [102]:
sad = len(rawDF[rawDF['What feelings are you having that makes you want to exercise?'].str.contains('Sad')])

sad

10

In [103]:
(sad / rawDFNum) * 100

31.25

31.25% of people feel like being sad makes them want to exercise.

In [104]:
angry = len(rawDF[rawDF['What feelings are you having that makes you want to exercise?'].str.contains('Angry')])

angry

11

In [106]:
(angry / rawDFNum) * 100

34.375

34.375% of people feel like being angry makes them want to exercise.

In [107]:
calm = len(rawDF[rawDF['What feelings are you having that makes you want to exercise?'].str.contains('Calm')])

calm

13

In [108]:
(calm / rawDFNum) * 100

40.625

40.625% of people feel like being calm makes them want to exercise.

Q7: What percentage listens to music, doesn't listen to music, and sometimes listens to music?

In [61]:
set( rawDF["Do you listen to music when you exercise?"].tolist() )

{'No', 'Sometimes', 'Yes'}

In [62]:
noMusic =  len(rawDF[rawDF["Do you listen to music when you exercise?"] == "No"])

noMusic

1

In [63]:
(noMusic / rawDFNum) * 100

3.125

3.125% of people doesn't listen to music.

In [64]:
yesMusic =  len(rawDF[rawDF["Do you listen to music when you exercise?"] == "Yes"])

yesMusic

23

In [65]:
(yesMusic / rawDFNum) * 100

71.875

71.9% of people do listen to music when they exercise.

In [66]:
someMusic =  len(rawDF[rawDF["Do you listen to music when you exercise?"] == "Sometimes"])

someMusic

8

In [67]:
(someMusic / rawDFNum) * 100

25.0

25.0% of people sometimes listen to music when they exercise.

Q8: What percentage of people chose a specific reason why they exercise at the specific locations?

In [110]:
access = len(rawDF[rawDF['Why do you exercise at the chosen location?'].str.contains('Easily Accessible')])

access

25

In [111]:
(access / rawDFNum) * 100

78.125

78.125% of people chose the specific locations because they were easily accessible.

In [114]:
equip = len(rawDF[rawDF['Why do you exercise at the chosen location?'].str.contains('Has equipment')])

equip

21

In [115]:
(equip / rawDFNum) * 100

65.625

65.625% of people chose the specific locations because they have equiptemnt.

In [117]:
scene = len(rawDF[rawDF['Why do you exercise at the chosen location?'].str.contains('Great Scenery')])

scene

11

In [118]:
(scene / rawDFNum) * 100

34.375

34.375% of people chose the specific locations because they have great scenery.

In [119]:
friends = len(rawDF[rawDF['Why do you exercise at the chosen location?'].str.contains('Friends exercise with you at the location')])

friends

18

In [120]:
(friends / rawDFNum) * 100

56.25

56.25% of people chose the specific locations because friends can exercise with you at the locations.

In [121]:
vibes = len(rawDF[rawDF['Why do you exercise at the chosen location?'].str.contains('Good vibes')])

vibes

14

In [122]:
(vibes / rawDFNum) * 100

43.75

43.75% of people chose the specific locations because they have good vibes.

Q9: What is percentage of people that thinks music helps with negative feelings?

In [68]:
helpsNeg =  len(rawDF[rawDF["Do you think listening to music while exercising helps relieve the negative feelings?"] == "Yes"])

helpsNeg

28

In [69]:
(helpsNeg / rawDFNum) * 100

87.5

87.5% of people believes music helps with negative feelings.

In [70]:
notHelpsNeg =  len(rawDF[rawDF["Do you think listening to music while exercising helps relieve the negative feelings?"] == "No"])

notHelpsNeg

4

In [71]:
(notHelpsNeg / rawDFNum) * 100

12.5

12.5% of people believes music doesn't help with negative feelings.